src/emqx_broker.erl

```diff
@@ -498,4 +534,36 @@ code_change(_OldVsn, State, _Extra) ->
 %%--------------------------------------------------------------------
 %% Internal functions
 %%--------------------------------------------------------------------
+sub_pre_check(Pid) ->
+    is_pid_links_port(Pid).
+
+-spec is_pid_links_port(pid()) -> boolean().
+is_pid_links_port(Pid) ->
```

Comment:
this might return false negative for none-tcp clients.

I need to find a new way to find the unexpected offline call.